spring-data-rest-tests/spring-data-rest-tests-jpa/src/test/java/org/springframework/data/rest/webmvc/jpa/CorsIntegrationTests.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.rest.webmvc.jpa;
+
+import static org.hamcrest.Matchers.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import org.junit.Test;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.rest.core.config.RepositoryRestConfiguration;
+import org.springframework.data.rest.tests.AbstractWebIntegrationTests;
+import org.springframework.data.rest.webmvc.BasePathAwareController;
+import org.springframework.data.rest.webmvc.RepositoryRestController;
+import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer;
+import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurerAdapter;
+import org.springframework.hateoas.Link;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Web integration tests specific to Cross-origin resource sharing.
+ * 
+ * @author Mark Paluch
+ * @soundtrack 2 Unlimited - No Limit
+ */
+@ContextConfiguration
+public class CorsIntegrationTests extends AbstractWebIntegrationTests {
+
+	static class CorsConfig extends JpaRepositoryConfig {
+
+		@Bean
+		RepositoryRestConfigurer repositoryRestConfigurer() {
+
+			return new RepositoryRestConfigurerAdapter() {
+
+				@Override
+				public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config) {
+
+					config.addCorsMapping("/books/**") //
+							.allowedMethods("GET", "PUT", "POST") //
+							.allowedOrigins("http://far.far.away");
+				}
+			};
+		}
+	}
+
+	@RepositoryRestController
```

Comment:
Let's move those classes under the test methods so that we stay consistent to other tests. Configuration can stay here.